### PR TITLE
Create permissions for acceptance tests

### PR DESCRIPTION
**Pull request by jaucourt**
https://eaflood.atlassian.net/browse/IWTF-1467

New functionality to create permission data structures for testing purposes

### CODE REVIEW

**Review by sgardnerdell — COMMENTED** [review]
Looks good.  Couple of comments but mostly aesthetic.  Does this work when referenced from a feature file?

File: packages/frontend-acceptance-tests/src/lib/createPermissions.js [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+const getGlobalOptionSetValue = async (name, lookup) => {
+  const llookup = lookup && lookup.toLowerCase()
+  const definition = await retrieveGlobalOptionSets().cached()
+  return definition[name] && lookup
+    ? Object.values(definition[name].options).find(o => o.label.toLowerCase() === llookup || o.description.toLowerCase() === llookup)
+    : undefined
+}
+
+const getEndDate = expiryDateSpec => {
+  const today = new Date()
```
[the remark this answers is not in the document]
Comment (sgardnerdell) [reply packages/frontend-acceptance-tests/src/lib/createPermissions.js:36]: On seconds thoughts - might this fail when using yesterday on the 1st of the month or tomorrow on the last day of the month?

File: package.json [hunk 2]
```diff
@@ -53,5 +54,8 @@
     "*": [
       "prettier-standard --lint"
     ]
+  },
```
[the remark this answers is not in the document]
Comment (jaucourt) [reply package.json:57]: It's actually used by the `pocl-test-file-generator` package as well, so it avoids duplication if it's in the root `package.json`

File: packages/frontend-acceptance-tests/src/lib/createPermissions.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (jaucourt) [reply packages/frontend-acceptance-tests/src/lib/createPermissions.js]: Nope, reinstated the tests and it works perfectly

File: packages/frontend-acceptance-tests/src/lib/createPermissions.js [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+  permission.referenceNumber = generateReferenceNumber(new Date(permission.endDate))
+  const licensee = await getContact()
+  permission.licensee = licensee
+  const permit = await getPermit()
+  permission.permitId = permit.id
+  permission.bindToEntity(Permission.definition.relationships.licensee, licensee)
+  permission.bindToEntity(Permission.definition.relationships.permit, permit)
+
+  // persist
+  persist(...[licensee, permission])
```
[the remark this answers is not in the document]
Comment (jaucourt) [reply packages/frontend-acceptance-tests/src/lib/createPermissions.js]: Removed

**Comment by andrewhick** [comment]
Just checked the Jira ticket for this one (IWTF-1467) and it's currently on hold., awaiting inclusion into an upcoming sprint.

**Pull request closed without merging**